### PR TITLE
dev/core#2397 Add readonly attribute to mostly modified_date fields a…

### PR DESCRIPTION
**Pull request by seamuslee001**
…nd some other fields that shouldn't really be updated

Overview
----------------------------------------
This adds the readonly attribute to a few more fields

Before
----------------------------------------
Fields not marked as readonly

After
----------------------------------------
Fields marked as readonly

@colemanw would appreciate your thoughts on these fields as a starting point

### CODE REVIEW

**Comment by colemanw** [comment]
retest this please

File: CRM/Activity/DAO/Activity.php [hunk 1]
```diff
@@ -524,6 +525,7 @@ public static function &fields() {
           'html' => [
             'label' => ts("Parent Activity"),
           ],
+          'readonly' => TRUE,
```
Comment (colemanw) [inline CRM/Activity/DAO/Activity.php:528]: Yes to `'parent_id'` being readonly. I would also add activity `'original_id'`.

File: CRM/Contact/DAO/Contact.php [hunk 2]
```diff
@@ -833,6 +833,7 @@ public static function &fields() {
             'keyColumn' => 'name',
             'optionEditPath' => 'civicrm/admin/options/languages',
           ],
+          'readonly' => TRUE,
```
Comment (colemanw) [inline CRM/Contact/DAO/Contact.php:836]: I'm not sure about  `'preferred_language'`. AFAIK CiviCRM doesn't do anything clever to set this value programatically?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 @colemanw I think we merged a replacement?

**Comment by colemanw** [comment]
Whoops, I think we both forgot about this PR and then I did that other one that's kind of but not totally redundant with this.

**Pull request closed without merging**